core/modules.rs

```diff
@@ -246,23 +259,78 @@ impl ModuleInfo {
   }
 }
 
+/// A symbolic module entity.
+pub enum SymbolicModule {
+  /// This module is an alias to another module.
+  /// This is useful such that multiple names could point to
+  /// the same underlying module (particularly due to redirects).
+  Alias(String),
+  /// This module associates with a V8 module by id.
+  Mod(deno_mod),
+}
```

Comment:
It's not clear to me that this is needed in core... But I really want to get a low-level test (which doesn't actually perform I/O - using MockLoader) to demonstrate how it fails.

Reply:
(Looks like no body is taking on this... In that case I'll continue working on it such that it does not break `deno info` in the next week's demo)

Basically multiple names can point to the exactly same compiled module due to redirection headers. Keeping this information as an enum also allows better later inspection. This feature would also be necessary in case we later want to introduce `import-maps` -- essentially adding more aliasing functionalities

Reply:
@kevinkassimo I’m working on it (slowly). It’s not clear to me that core/modules.rs needs any changes. I’m trying to be careful not to introduce unnecessary machinery in core ... what’s your thought? Are the core changes needed?

Reply:
Depends on how likely users of the crate would need this feature...

If we were to move the aliasing functionality to `cli` we might need to make `Modules` more of a trait so that in `cli` we can provide our own implementation has provides the feature, as at this moment `RecursiveLoad` relies directly on the `Modules` living on the `Loader`.

I do not really have a full clear understanding on the `core` crate so my judgment might be wrong (also kind of reluctant to change the code structure too heavily without coordinations).

Anyways, I've just push my code with a test added, and if this does not seem to be the best way to address the issue, it could still serve as a reference about how to potentially fix the problem

Reply:
There is indeed an option to change not a single line in `core/module.rs`: add some tracking mechanism in `cli/worker.rs` and modify impl `load()` to apply a mapping of paths before & after redirect and supply the corresponding code. In that case the fact that source code is served from a different url is transparent to the `core`... Although I'm not so sure about the cost of implementing this (and we will need to take care of some really weird corner cases)

Reply:
It does seem like it would be easier to put the aliasing in core. But my concern is: is this the only addition that is needed? Or will we start adding extra bits for various corner cases? I can’t think of any more exceptions right now - if that held I think aliasing in core is the right way to go. But if it turns out there will be many - the extra machinery in CLI to map paths might be better...

Reply:
Yeah I agree. I currently also cannot think of a corner case that might break this implementation yet...

As to using a cli only mapping, this actually requires some extra machinery: we will also need `resolve()` to access `Loader` members, making it no longer a static method (as a `load()`-only tweak would immediately break on cases like "../a.ts" relative to "https://example.com" (which redirects to "https://github.com/some_repo/index.ts", as "https://github.com/a.ts" makes sense but not "http://a.ts")).

Reply:
> we will also need resolve() to access Loader members, making it no longer a static method 

This I’d very much like to avoid - because currently the resolve_module_spec function is exactly (I think?) as specified in the ES module spec.

Reply:
Ok - I think the patch is quite clean now. I will land it - it seems like the best solution. It's still isolated from core/isolate.rs - so the simple module layer that V8 presents is still accessible. Please just make one change where I commented "unreachable!()"